File: tests/fixtures/openai-realtime/openai-realtime-no-log-raw-message-payloads-fixed/02-adversarial-non-openai-socket.ts
import WebSocket from 'ws';

// Adversarial: a different WebSocket (not the OpenAI Realtime connection)
// logs its raw message verbatim. Out of scope for this rule.
function setup(logger: any) {
  const internalSocket = new WebSocket('wss://internal.example.com/events');

  internalSocket.on('message', (msg) => {
    logger.info(`Internal event: ${msg}`);
  });
}

File: tests/fixtures/openai-realtime/openai-realtime-no-log-raw-message-payloads-fixed/02_adversarial_non_openai_socket.py
import logging

import websockets

# Adversarial: a different WebSocket entirely, not the OpenAI Realtime API.
OTHER_URL = "wss://example.com/v1/stream"
logger = logging.getLogger(__name__)


async def stream():
    async with websockets.connect(OTHER_URL) as ws:
        async for message in ws:
            logger.info(message)
